packages/app/components/token-picker/TokenPickerListRow.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import { BodyText } from "@/ui";
+import { TokenIcon } from "../TokenIcon";
+import { Token } from "@/models/token/types";
+
+interface TokenPickerListRowProps {
+  token: Token;
+  closeAction: () => void;
+}
+
+export const TokenPickerListRow = ({
+  closeAction,
+  token
+}: TokenPickerListRowProps) => (
+  <div
+    className="flex justify-between w-full px-6 py-2 cursor-pointer hover:bg-surface-50"
+    key={`${token.address}${token.symbol}`}
```

Comment:
I think address is enough for the key. But I'm okay keeping it like it is if you want.

```suggestion
    key={token.address}
```

Reply:
At least on the provided `tokenList` from the custom hook, we have different tokens with the same address. I did it this way as quick fix for that, but didn't check WHY do we have repeated addies. Maybe we have duplicated tokens. Will further check that and see what's happening

Reply:
That's odd. Thanks for checking that out.

Reply:
```json
    {
        "name": "DeHive",
        "address": "0xFbdd194376de19a88118e84E279b977f165d01b8",
        "symbol": "DHV",
        "decimals": 18,
        "chainId": 100,
        "logoURI": "https://dehive.finance/logo_200x200.png"
    },
    {
        "name": "beefy.finance",
        "address": "0xFbdd194376de19a88118e84E279b977f165d01b8",
        "symbol": "BIFI",
        "decimals": 18,
        "chainId": 137,
        "logoURI": "https://raw.githubusercontent.com/beefyfinance/beefy-app/prod/src/images/single-assets/BIFI.png"
    },
```

So basically we have same addy but different chain ID. Should I keep maybe ```key={`${token.address}${token.chainId}`}``` instead?

Reply:
if symbol is different we can use the one you used above or maybe we can add a dash to it. Up to you.

```js
key={`${token.symbol}-${token.address}`}
```

Reply:
IMO we shouldn't have tokens with chainId 137

Reply:
Only chainId 1 and 100, those are the ones we are using and defining in the app.

Reply:
As per an offline talk with @berteotti , we concluded there are 2 different things happening here:
* `TokenPicker` token list should only display tokens for the connected wallet's current chain (be it Gnosis or Mainnet), so only display tokens from 1 chain at a time. -> Hence we can use `token.address` as ID.
* There's a bug right now where `tokenList` returns multiple tokens from multiple chains (and some tokens without ChainId), this will be addressed in https://linear.app/swaprdev/issue/STK-93/tokenlist-has-tokens-from-unsupported-chains

Reply:
The solution looks good for me.  We can move on with that.

Reply:
Do you know which token list the chainId 137 came from?

Reply:
Sounds good to me.

Reply:
> Do you know which token list the chainId 137 came from?

No, sorry didn't try to find out that, decided to let that as work to do for STK-93